lib/chef/resource/windows_task.rb

```diff
@@ -49,67 +49,118 @@ class WindowsTask < Chef::Resource
                                       :on_logon,
                                       :onstart,
                                       :on_idle,
-                                      :none], default: :hourly
+                                      :none]
       property :start_day, String
       property :start_time, String
       property :day, [String, Integer]
       property :months, String
       property :idle_time, Integer
       property :random_delay, [String, Integer]
       property :execution_time_limit, [String, Integer], default: "PT72H" # 72 hours in ISO8601 duration format
+      property :minutes_duration, [String, Integer]
+      property :minutes_interval, [String, Integer]
 
-      attr_accessor :exists, :status, :enabled
+      attr_accessor :exists, :task
+
+      SYSTEM_USERS = ['NT AUTHORITY\SYSTEM', "SYSTEM", 'NT AUTHORITY\LOCALSERVICE', 'NT AUTHORITY\NETWORKSERVICE', 'BUILTIN\USERS', "USERS"].freeze
+      VALID_WEEK_DAYS = %w{ mon tue wed thu fri sat sun * }
+      VALID_DAYS_OF_MONTH = ("1".."31").to_a << "last" << "lastday"
+      VALID_MONTHS = %w{JAN FEB MAR APR MAY JUN JUL AUG SEP OCT NOV DEC *}
+      VALID_WEEKS = %w{FIRST SECOND THIRD FOURTH LAST LASTDAY}
 
       def after_created
         if random_delay
           validate_random_delay(random_delay, frequency)
-          duration = sec_to_dur(random_delay)
-          random_delay(duration)
+          random_delay(sec_to_min(random_delay))
         end
 
         if execution_time_limit
-          unless execution_time_limit == "PT72H" # don't double convert an ISO8601 format duration
-            raise ArgumentError, "Invalid value passed for `execution_time_limit`. Please pass seconds as an Integer (e.g. 60) or a String with numeric values only (e.g. '60')." unless numeric_value_in_string?(execution_time_limit)
-            duration = sec_to_dur(execution_time_limit)
-            execution_time_limit(duration)
-          end
+          execution_time_limit(259200) if execution_time_limit == "PT72H"
+          raise ArgumentError, "Invalid value passed for `execution_time_limit`. Please pass seconds as an Integer (e.g. 60) or a String with numeric values only (e.g. '60')." unless numeric_value_in_string?(execution_time_limit)
+          execution_time_limit(sec_to_min(execution_time_limit))
         end
 
+        # Removed the default value :hourly for frequency in Chef 14.1 following code added for backward compatibility
+        chef_version = ::Chef::VERSION.split(".")
+        if chef_version[0].to_i > 14 || (chef_version[0].to_i == 14 && chef_version[1].to_i > 0)
```

Comment:
We need to remove this version check and go one way or another here. Otherwise we're testing behavior that will be different on release.